tests/unit/supplements/factory_supplements_test.py

```diff
@@ -105,15 +105,18 @@ def foobar(no_default):
         with pytest.raises(TypeError):
             get_user_defined_mapping()['a']['opid']()
 
-    # We should never use the endpoint-specific fixture.
     def test_nested_dependency(self):
         assert get_user_defined_mapping()['nested_dependency']['opid']() == 4
-        assert get_user_defined_mapping()['nested_dependency']['new_opid']() == 4
 
-    # If the dependency is endpoint-specific then we should fail.
-    def test_endpoint_dependency_fails(self):
-        with pytest.raises(TypeError):
-            get_user_defined_mapping()['caller']['only_opid']()
+    # We resolve operation_id-specific dependencies if the current
+    # factory is specified for the same operation id.
+    def test_endpoint_dependency_single(self):
+        get_user_defined_mapping()['caller']['only_opid']() == 3
+
+    # If the current factory is not specified for the same operation_id,
+    # then we only use the default values for nested dependencies.
+    def test_nested_endpoint_dependency_uses_default(self):
+        assert get_user_defined_mapping()['nested_dependency']['new_opid']() == 4
```

Comment:
- `nested_dependency` returns `caller * 2`.
- `caller` (with the `new_opid`) returns `special_caller`
- `special_caller` returns `dependency + 2`
- `dependency` uses the default registration and returns `1`.

Therefore, shouldn't this be 6?

```
(1 + 2) * 2
```